core/domain/translation_services.py

```diff
@@ -46,14 +45,6 @@ def get_and_cache_machine_translation(
     Returns:
         str|None. The translated text or None if no translation is found.
     """
-    translation = translation_fetchers.get_machine_translation(
-        source_language_code,
-        target_language_code,
-        source_text.strip()
-    )
-    if translation is not None:
-        return translation.translated_text
-
```

Comment:
Why remove the caching mechanism? Also, can you check the backend tests? I'm surprised `test_get_machine_translation_checks_datastore_first()` still passes.